Cost Differentiator

hug33k opened the pull request:
Added Cost Differentiator table in Cost Breakdown view.

You can select a timerange and an interval, it will automatically update data in table to show you cost and their variation month to month or week to week.

P.S. : Tests update is in progress

alexis-ld commented:
Looks good overall. Just some quick changes :
- Add a cost diff table by default on view (at second place)
- by default it shows useless data ($0 items) please order the table by most recent month cost descending by default
- please add a "+" before a positive value in the variation column for consistency
- for variation overs 100% (in either way) please apply a 'font-weight: 500' property on the value. This will make the big variation to stand out a bit more (_see Screenshot 1_)
- please fix your variation calculation, it struggles on 0 values (_see screenshot 2 & 3_)
- the available timerange presets are not useful at all for this element (last 7 days, ...) Please use more adapted presets for this particular component (Last 2 month (default), Last 3 months, Last 4 months,...)
- please change the top button to add this element to "Add a month-to-month table", "differentiatior" is an ugly word ;) 
- we now have a lot of different components on this view. it would be nice to have a small discrete title on the top of each panel (with an icon) just stating the current component (e.g Pie chart, Month-to-month table, Bar Chart, ...) we have empty space for this on the top left of each panel

Screenshot 1
![screen shot 2018-03-26 at 12 53 38](https://user-images.githubusercontent.com/3480526/37902813-edf672e8-30f5-11e8-9088-635b46616716.png)
Screenshot 2
![screen shot 2018-03-26 at 12 55 42](https://user-images.githubusercontent.com/3480526/37902811-eda24b64-30f5-11e8-8c5b-5a83c9b4e644.png)
Screenshot 3
![screen shot 2018-03-26 at 12 55 34](https://user-images.githubusercontent.com/3480526/37902812-edcce176-30f5-11e8-85e1-005e1f15e35b.png)

hug33k commented:
@alexis-ld :
- 1 : Updated, there is now 4 default charts.
- 2/3/4/5 : Updated
- 6 : I have updated default ranges for this component : Last 2/3 Weeks, This Month, Last Month, Last 2/3/6/12 Months, This Year, Last Year. (I had to keep weeks ranges since you can use `weeks` as interval.)
- 7 : Button name is now "Add a Cost Table"
- 8/9 : Updated

I still need to update tests but you can already review these changes.
